packages/tools/src/stateManagement/segmentation/SegmentationStateManager.ts

```diff
@@ -435,10 +435,10 @@ export default class SegmentationStateManager {
    */
   addColorLUT(colorLUT: ColorLUT, lutIndex: number): void {
     if (this.state.colorLUT[lutIndex]) {
-      console.log('Color LUT table already exists, overwriting');
+      console.warn('Color LUT table already exists, overwriting');
     }
 
-    this.state.colorLUT[lutIndex] = colorLUT;
+    this.state.colorLUT[lutIndex] = structuredClone(colorLUT);
```

Comment:
Cloning large objects - particularly if we need a 16 bit LUT, is fairly expensive, and is really only required when a value is changing.  Consider storing inside the object a value that says to take ownership of it, and that the value is not to be touched.  Then, if you apply the update to the LUT, go ahead and clone it at that time, only if it hasn't already been cloned.